db/helpers/botDataUtils.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * db/helpers/botDataUtils.js
+ *
+ * Used by the Bot Data model.
+ */
+const _ = require('lodash');
+const ZERO = 0;
+const ONE = 1;
+const TWO = 2;
+
+/**
+ * Check if a string is a JSON
+ *
+ * @param {String} item - String to check
+ * @returns {Boolean} - if String is JSON
+ */
+function isJson(item) {
```

Comment:
https://www.npmjs.com/package/is-json?

Reply:
I tried this package but it doesnt seem to be as good at the code i wrote it return false for this string

{ response: 'this is a {$TestBot1.response$}' }

Reply:
Is `{$varName$}` a standard for doing variable replacement? Where is this used? Why not more standard `${varName}` or "template" style `{{varName}}`?

Reply:
Some of the bots already use variable placements, like comms bot and pagerduty use handlebars. So we needed something unquie that would conflict with things

Reply:
Could be confusing, hard for bot developers to grok non-standard syntax. What alternatives have you explored?

Reply:
Ive decided to switch ${} since handlebars use {{}}; and I dont know another JS template that use it withouth ``